File: ios-bridge/ios-bridge/PlaneBox.m
//
//  PlaneBox.m
//  ios-bridge
//
//  Created by Fredrik on 31.03.18.
//  Copyright © 2018 fwcd. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "PlaneBox.h"

@implementation PlaneBox

- (id) initAtX:(float)x y:(float)y withWidth:(float)w andHeight:(float)h filled:(bool)filled {
    [self setX:x];
    [self setY:y];
    [self setW:w];
    [self setH:h];
    [self setFilled:filled];
    return self;
}

@end

File: ios-bridge/ios-bridge/PlaneBox.swift
//
//  PlaneBox.swift
//  ios-bridge
//
//  Created by Fredrik on 02.04.18.
//  Copyright © 2018 fwcd. All rights reserved.
//

import Foundation
import UIKit

@objc
public class PlaneBox: NSObject {
	public let x: Float
	public let y: Float
	public let w: Float
	public let h: Float
	public let filled: Bool
	public let color: UIColor
	
	@objc
	public init(atX x: Float, andY y: Float, withWidth w: Float, andHeight h: Float, filled: Bool, color: UIColor) {
		self.x = x
		self.y = y
		self.w = w
		self.h = h
		self.filled = filled
		self.color = color
	}
	
	public func toCGRect() -> CGRect {
		return CGRect(x: Double(x), y: Double(y), width: Double(w), height: Double(h))
	}
}
